librevna: init at 1.6.5

dunxen opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

dunxen committed:
librevna: init at 1.6.5

dunxen commented:
Thanks to @mtahernia for the prior effort on this. Please let me know if you are willing to be a maintainer for this package too.

sikmir commented on pkgs/by-name/li/librevna/package.nix:
```diff
@@ -0,0 +1,71 @@
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    cp LibreVNA-GUI $out/bin/
```
```suggestion
    install -Dm755 LibreVNA-GUI -t $out/bin
```

sikmir commented on pkgs/by-name/li/librevna/package.nix:
```diff
@@ -0,0 +1,71 @@
+  src = fetchFromGitHub {
+    owner = "jankae";
+    repo = "LibreVNA";
+    rev = "v${finalAttrs.version}";
```
```suggestion
    tag = "v${finalAttrs.version}";
```

sikmir commented:
You don't need explicit `buildPhase`:
```diff
diff --git i/pkgs/by-name/li/librevna/package.nix w/pkgs/by-name/li/librevna/package.nix
index a0f78c254d59..a95d23ac8b66 100644
--- i/pkgs/by-name/li/librevna/package.nix
+++ w/pkgs/by-name/li/librevna/package.nix
@@ -20,7 +20,12 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-3iQO4iUhRJWFqqrMMKXslPm8wtl+WaEa9HTrv5bKHJ4=";
   };

-  nativeBuildInputs = [ qt6.wrapQtAppsHook ];
+  sourceRoot = "${finalAttrs.src.name}/Software/PC_Application/LibreVNA-GUI";
+
+  nativeBuildInputs = [
+    qt6.qmake
+    qt6.wrapQtAppsHook
+  ];

   buildInputs = [
     qt6.qtbase
@@ -36,14 +41,6 @@ stdenv.mkDerivation (finalAttrs: {
     export QT_SELECT=qt6
   '';

-  buildPhase = ''
-    runHook preBuild
-    cd Software/PC_Application/LibreVNA-GUI
-    qmake LibreVNA-GUI.pro
-    make -j $NIX_BUILD_CORES
-    runHook postBuild
-  '';
-
   installPhase = ''
     runHook preInstall
     mkdir -p $out/bin
```

sikmir commented on pkgs/by-name/li/librevna/package.nix:
```diff
@@ -0,0 +1,69 @@
+  buildInputs = [
+    qt6.qtbase
+    libusb1
+    git
```
Is `git` really needed?

dunxen replied:
Oh nope. Thanks!

dunxen commented on pkgs/by-name/li/librevna/package.nix:
```diff
@@ -0,0 +1,69 @@
+    udev
+  ];
+
+  preConfigure = ''
```
Oh thanks, indeed with removing the explicit `buildPhase`, it doesn't keep falling back to C ANSI locale. Will remove this too.

dunxen commented on pkgs/by-name/li/librevna/package.nix:
```diff
@@ -0,0 +1,56 @@
+
+  postInstall = ''
+    install -Dm644 ../51-vna.rules $out/etc/udev/rules.d
+  '';
```
Ah, that's great. Thanks.

sikmir commented on patch.diff:
Remove patch please.

sikmir commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 446590`
Commit: [`273f044fbc072be257fe538d1f2e3afb8e2364ce`](https://github.com/NixOS/nixpkgs/commit/273f044fbc072be257fe538d1f2e3afb8e2364ce) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/273f044fbc072be257fe538d1f2e3afb8e2364ce..pull/446590/head))
Merge: [`ac4140ae70d38a3f42c097348b24b1571bca31b2`](https://github.com/NixOS/nixpkgs/commit/ac4140ae70d38a3f42c097348b24b1571bca31b2)

Logs: https://github.com/sikmir/nixpkgs-review-gha/actions/runs/18062826664


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>librevna</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>librevna</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>librevna</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>librevna</li>
  </ul>
</details>

dunxen commented:
Just removed the patch.diff file I accidentally commited :man_facepalming: 

<details>
<summary>`git diff-tree --minimal -U1 273f044 aaee767`</summary>

```diff
diff --git a/patch.diff b/patch.diff
deleted file mode 100644
index 8e8e84f7b4de..000000000000
--- a/patch.diff
+++ /dev/null
@@ -1,73 +0,0 @@
-diff --git i/pkgs/by-name/li/librevna/package.nix w/pkgs/by-name/li/librevna/package.nix
-index e6d607e36da7..1e830b003118 100644
---- i/pkgs/by-name/li/librevna/package.nix
-+++ w/pkgs/by-name/li/librevna/package.nix
-@@ -2,6 +2,8 @@
-   stdenv,
-   lib,
-   fetchFromGitHub,
-+  apple-sdk_14,
-+  pkg-config,
-   qt6,
-   libusb1,
-   udev,
-@@ -21,6 +23,7 @@ stdenv.mkDerivation (finalAttrs: {
-   sourceRoot = "${finalAttrs.src.name}/Software/PC_Application/LibreVNA-GUI";
-
-   nativeBuildInputs = [
-+    pkg-config
-     qt6.qmake
-     qt6.wrapQtAppsHook
-   ];
-@@ -28,17 +31,32 @@ stdenv.mkDerivation (finalAttrs: {
-   buildInputs = [
-     qt6.qtbase
-     libusb1
--    udev
--  ];
-+  ]
-+  ++ lib.optional stdenv.hostPlatform.isLinux udev
-+  ++ lib.optional stdenv.hostPlatform.isDarwin apple-sdk_14;
-
--  installPhase = ''
--    runHook preInstall
--    install -Dm755 LibreVNA-GUI -t $out/bin
--    install -Dm644 ../51-vna.rules -t $out/etc/udev/rules.d
--    install -Dm644 resources/librevna.desktop -t $out/share/applications
--    install -Dm644 resources/librevna.png -t $out/share/pixmaps
--    runHook postInstall
--  '';
-+  installPhase =
-+    if stdenv.hostPlatform.isDarwin then
-+      ''
-+        runHook preInstall
-+
-+        mkdir -p $out/{Applications,bin}
-+        mv LibreVNA-GUI.app $out/Applications
-+        ln -s $out/Applications/LibreVNA-GUI.app/Contents/MacOS/LibreVNA-GUI $out/bin/LibreVNA-GUI
-+
-+        runHook postInstall
-+      ''
-+    else
-+      ''
-+        runHook preInstall
-+
-+        install -Dm755 LibreVNA-GUI -t $out/bin
-+        install -Dm644 ../51-vna.rules -t $out/etc/udev/rules.d
-+        install -Dm644 resources/librevna.desktop -t $out/share/applications
-+        install -Dm644 resources/librevna.png -t $out/share/pixmaps
-+
-+        runHook postInstall
-+      '';
-
-   meta = {
-     changelog = "https://github.com/jankae/LibreVNA/releases/tag/v${finalAttrs.version}";
-@@ -46,7 +64,7 @@ stdenv.mkDerivation (finalAttrs: {
-     description = "100kHz to 6GHz 2 port USB based VNA";
-     mainProgram = "LibreVNA-GUI";
-     license = lib.licenses.gpl3;
--    platforms = lib.platforms.linux;
-+    platforms = lib.platforms.unix;
-
-     maintainers = with lib.maintainers; [
-       dunxen

```

</details>

felbinger commented on pkgs/by-name/li/librevna/package.nix:
```diff
@@ -0,0 +1,79 @@
+stdenv.mkDerivation (finalAttrs: {
+  pname = "librevna";
+  version = "1.6.5";
+
```
```suggestion
__structuredAttrs = true;

```

kirillrdy commented on pkgs/by-name/li/librevna/package.nix:
why not --replace-fail ?

dunxen replied:
Yeah, that should indeed be a fail. Thanks
